ckanext/useraffiliation/actions.py

```diff
@@ -0,0 +1,109 @@
+import copy
+import datetime
+import logging
+from ckan import model, logic
+from ckan.plugins import toolkit
+
+log = logging.getLogger(__name__)
+
+
+def _get_user_obj(context):
+    if "user_obj" in context:
+        return context["user_obj"]
+    user = context.get("user")
+    m = context.get("model", model)
+    user_obj = m.User.get(user)
+    if not user_obj:
+        raise toolkit.ObjectNotFound("User not found")
+    return user_obj
+
+
+@toolkit.chained_action
+def user_show(original_action, context, data_dict):
+    user = original_action(context, data_dict)
+    user_obj = _get_user_obj(context)
+
+    extras = _init_plugin_extras(user_obj.plugin_extras)
+    extras = _validate_plugin_extras(extras["useraffiliation"])
+
+    user["job_title"] = extras["job_title"]
+    user["affiliation"] = extras["affiliation"]
+
+    return user
+
+
+@toolkit.chained_action
+def user_create(original_action, context, data_dict):
+    user = original_action(context, data_dict)
+    user_obj = _get_user_obj(context)
+
+    if not data_dict.get("job_title"):
+        raise toolkit.ValidationError({"job_title": ["A job title must be provided."]})
+
+    if not data_dict.get("affiliation"):
+        raise toolkit.ValidationError(
+            {"affiliation": ["An affiliation must be provided."]}
+        )
+
+    plugin_extras = _init_plugin_extras(user_obj.plugin_extras)
+    plugin_extras["useraffiliation"]["affiliation"] = data_dict["affiliation"]
+    plugin_extras["useraffiliation"]["job_title"] = data_dict["job_title"]
+    user_obj.plugin_extras = plugin_extras
+
+    if not context.get("defer_commit"):
+        m = context.get("model", model)
+        model.Session.commit()
```

Comment:
:+1: This makes sense.  Sometimes you may want to call the action to "create a user" but not actually commit any changes to the db.  The context param "defer_commit" is used to indicate this in core ckan. 

But I think I might do it like this...
```suggestion
    if not context.get("defer_commit"):
        context['model'].Session.commit()
```

There's actually a context['session'] object that should in theory be populated.  IAgain I'm not sure why he isn't using it :thinking: